H2 optimize tests

pjstanle opened the pull request:
Gradient-free H2 optimization scripts running with most recent changes. Test for optimization script.

dguittet commented:
This test has revealed that h2_setup_optimize.py has a seg fault on Linux / HPC. It ran fine on Mac though.

```
(/home/dguittet/.conda-envs/hybrid) [dguittet@el3 H2_Analysis]$ python h2_setup_optimize.py 
Segmentation fault (core dumped)
```

dguittet commented:
The seg faults may have to do with battery sizing where the power capacity is 0 but the energy is non-zero. This doesn't really make sense model-wise. Is the battery duration a fixed or a design variable?

dguittet commented on examples/H2_Analysis/h2_setup_optimize.py:
```diff
@@ -287,91 +78,79 @@ def setup_cost_calcs(scenario,hybrid_plant,electrolyzer_size_mw,wind_size_mw,sol
+        interconnection_size = electrolyzer_size * 1000.0
     wind_size_mw = scenario['Turbine Rating']
-    hybrid_plant = setup_power_calcs(scenario,wind_size_mw,solar_capacity_mw,battery_storage_mwh,battery_discharge_rate) 
+    hybrid_plant = setup_power_calcs(scenario,wind_size_mw,solar_capacity_mw,battery_storage_mwh,battery_discharge_rate,interconnection_size) 
```
The function parameters for setup_power_calcs is:

```
setup_power_calcs(scenario,wind_size_mw,solar_size_mw,storage_size_mwh,storage_size_mw,interconnection_size_mw)
```

but here you have battery_discharge_rate instead of storage_size_mw

pjstanle replied:
Yes this is fine for now. Leaving open the option of having different charge and discharge rates in the future.

pjstanle commented:
I added doc strings to all of the functions in the optimization files. I also added a flag to check for zero battery capacity and changed the bounds in the GF optimization. Hopefully that will fix the seg fault issue.
